plugins/device/rest.go

```diff
@@ -93,29 +93,47 @@ func deviceAddHandler(w http.ResponseWriter, r *http.Request) {
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "transaction to prepare device failed")
 		return
 	}
-	peerInfo, err = peer.GetPeer(peerID)
+	deviceInfo, err := deviceutils.GetDevice(peerID, req.Device)
 	if err != nil {
-		logger.WithError(err).WithField("peerid", peerID).Error("Failed to get peer from store")
-		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get peer from store")
+		logger.WithError(err).WithField("peerid", peerID).Error("Failed to get device from store")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, "failed to get device from store")
 		return
 	}
 
-	// FIXME: Change this to http.StatusCreated when we are able to set
-	// location header with a unique URL that points to created device.
-	restutils.SendHTTPResponse(ctx, w, http.StatusOK, peerInfo)
+	restutils.SetLocationHeader(r, w, strings.TrimLeft(deviceInfo.Device, "/"))
+	restutils.SendHTTPResponse(ctx, w, http.StatusCreated, deviceInfo)
 }
 
 func deviceListHandler(w http.ResponseWriter, r *http.Request) {
 
 	ctx := r.Context()
 	logger := gdctx.GetReqLogger(ctx)
 	peerID := mux.Vars(r)["peerid"]
-	if uuid.Parse(peerID) == nil {
+	if peerID != "" && uuid.Parse(peerID) == nil {
 		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "invalid peer-id passed in url")
 		return
 	}
 
-	devices, err := deviceutils.GetDevices(peerID)
+	devName := mux.Vars(r)["device"]
+	if devName != "" && !strings.HasPrefix(devName, "/") {
```

Comment:
do we require prefix check here?

Reply:
realized that not required, observed that URLs will not have double `/` when it reaches here.